src/PowerShellEditorServices/Services/TextDocument/ScriptFile.cs

```diff
@@ -411,7 +411,7 @@ public void ApplyChange(FileChange fileChange)
         /// <returns>The zero-based offset for the given file position.</returns>
         public int GetOffsetAtPosition(int lineNumber, int columnNumber)
         {
-            Validate.IsWithinRange("lineNumber", lineNumber, 1, this.FileLines.Count);
+            Validate.IsWithinRange("lineNumber", lineNumber, 1, this.FileLines.Count + 1);
```

Comment:
Maybe this should just be a separate check for `FileLines.Count == 0`? I feel like the check is correct here, it just doesn't account for empty? Maybe it should be `Math.Max(this.FileLines.Count, 1)`?

Reply:
The use of `IsWithinRange` does not work well with 1-indexed lists that could be empty. I agree this looks weird, but it's what's done here too:

https://github.com/PowerShell/PowerShellEditorServices/blob/149f5ac1089fef8344f8a5849a211a1ba9a88431/src/PowerShellEditorServices/Services/TextDocument/ScriptFile.cs#L230-L237